### PR TITLE
Fix Unit Tests and Add Github Actions

### PR DESCRIPTION
**Pull request by ddasilva**
The merge request fixes the unit tests, adds a few stubs for the `make_levelN.py` scripts, and adds Github actions integration so they will be run automatically when a pull request is made. The README is updated to tell users how to run the unit tests.

I noticed that the `requirements.txt` only works with Python 3.9 and 3.10 (not 3.11 or 3.12) because the pinned packages aren't published for those Python versions. I put a note in the README and setup the github actions to only use 3.9 and 3.10.

image

### CODE REVIEW

File: suncet_processing_pipeline/make_level1.py [hunk 1]
```diff
@@ -127,19 +125,21 @@ def __badpixel_removal(self, input_map, badpixel_path):
...
             input_map (sunpy.map): Input 2D map (image) containing pixel values.
-            outlier_mask (numpy.ndarray): Boolean mask indicating bad pixels (True for bad pixels, False for good pixels).
 
         Returns:
             sunpy.map: Interpolated sunpy map with bad pixels replaced by interpolated values.
         """
         if self.config.cosmic_ray_removal:
+
+            outlier_mask = utilities.detect_outliers(input_map.data, 500)
+
```
Comment (ddasilva) [inline suncet_processing_pipeline/make_level1.py:142]: I'm not sure where this is coming from-- not me. Do you recognize this?

**Comment by ddasilva** [comment]
Someone with admin privileges to the repo might have to enable Github actions:
https://docs.github.com/en/repositories/managing-your-repositorys-settings-and-features/enabling-features-for-your-repository/managing-github-actions-settings-for-a-repository

File: suncet_processing_pipeline/make_level1.py (hunk 1 above)
Comment (starfleetjames) [reply suncet_processing_pipeline/make_level1.py:142]: That looks like a @MSKirk line

Comment (MSKirk) [reply suncet_processing_pipeline/make_level1.py:142]: yep that is me. is that causing you headaches?

Comment (ddasilva) [reply suncet_processing_pipeline/make_level1.py:142]: I'm not sure how it got in this pull request-- did someone force push an old version over main? Is it bad if we let this sneak in with this PR? it seems harmless

Comment (MSKirk) [reply suncet_processing_pipeline/make_level1.py:142]: It looks like it was gobbled up from some commits that I did last week. I am not sure how it got aggregated into this PR. If it looks alright to you, we can include it. I think that is easier than trying to separate it back out again just to merge it.

**Review by MSKirk — APPROVED** [review]
Looks good. @ddasilva is already improving our processes.

**Pull request merged**